### PR TITLE
fix: make redis cache pool settings configurable

### PR DESCRIPTION
**Pull request by leleueri**
related to AM-3974

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `6.4.4-am-3974-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/node/gravitee-node/6.4.4-am-3974-SNAPSHOT/gravitee-node-6.4.4-am-3974-SNAPSHOT.zip)

### CODE REVIEW

File: gravitee-node-cache/gravitee-node-cache-plugin-redis/src/main/java/io/gravitee/node/plugin/cache/redis/configuration/RedisOptionsFactory.java [hunk 1]
```diff
@@ -71,8 +75,13 @@ private static RedisOptions buildRedisOptions(RedisConfiguration configuration)
...
 
-        // Set max waiting handlers high enough to manage high throughput since we are not using the pooled mode
-        options.setMaxWaitingHandlers(1024);
+        options.setPoolName("cache-redis-" + UUID.randomUUID());
+        ofNullable(configuration.getMaxPoolSize()).ifPresent(options::setMaxPoolSize);
+        ofNullable(configuration.getMaxPoolWaiting()).ifPresent(options::setMaxPoolWaiting);
+        ofNullable(configuration.getPoolCleanerInterval()).ifPresent(options::setPoolCleanerInterval);
+        ofNullable(configuration.getPoolRecycleTimeout()).ifPresent(options::setPoolRecycleTimeout);
+        ofNullable(configuration.getMaxWaitingHandlers()).ifPresent(options::setMaxWaitingHandlers);
+
```
Comment (guillaumelamirand) [inline gravitee-node-cache/gravitee-node-cache-plugin-redis/src/main/java/io/gravitee/node/plugin/cache/redis/configuration/RedisOptionsFactory.java:84]: Shouldn't we set some default value for all of those properties ?

**Review by guillaumelamirand — CHANGES_REQUESTED** [review]
Could you add documentation to those configuration ?

File: gravitee-node-cache/gravitee-node-cache-plugin-redis/src/main/java/io/gravitee/node/plugin/cache/redis/configuration/RedisOptionsFactory.java (hunk 1 above)
Comment (leleueri) [reply gravitee-node-cache/gravitee-node-cache-plugin-redis/src/main/java/io/gravitee/node/plugin/cache/redis/configuration/RedisOptionsFactory.java:84]: RedisOption already provide default values, that's why I didn't provide ones and rely on redis.

**Pull request merged**